plugins/wasm-rust/src/dispatcher.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::hostcalls;
+use crate::promise::Promise;
+use std::collections::HashMap;
+use std::rc::Rc;
+use std::time::Duration;
+
+#[derive(Default)]
+pub struct HttpDispatcher {
+    m: HashMap<u32, Rc<Promise<(u32, usize, usize, usize)>>>,
+}
+
+impl HttpDispatcher {
+    pub fn dispatch(
+        &mut self,
+        upstream: &str,
+        headers: Vec<(&str, &str)>,
+        body: Option<&[u8]>,
+        trailers: Vec<(&str, &str)>,
+        timeout: Duration,
+    ) -> Rc<Promise<(u32, usize, usize, usize)>> {
```

Comment:
Based on this, further encapsulation is needed, such as supporting the extraction of domain and path from URL and cluster, as well as supporting get/post, etc